app/src/components/domain/RiskImminentEvents/WfpAdam/index.tsx

```diff
@@ -83,6 +117,46 @@ function WfpAdam(props: Props) {
         apiType: 'risk',
         url: '/api/v1/adam-exposure/{id}/exposure/',
         pathVariables: ({ eventId }) => ({ id: Number(eventId) }),
+        onSuccess: (res) => {
+            const { storm_position_geojson: stormPositionGeoJson } = res;
+
+            const stormPositions = isValidFeatureCollection(stormPositionGeoJson)
+                ? stormPositionGeoJson
+                : undefined;
+
+            const updatedLayers = { ...defaultLayersValue };
+
+            stormPositions?.features?.find((feature) => {
+                if (feature?.geometry.type === 'Point' || feature?.geometry.type === 'MultiPoint') {
+                    updatedLayers[LAYER_CYCLONE_NODES] = true;
+                }
+                return undefined;
+            });
+
+            stormPositions?.features?.find((feature) => {
+                if (feature?.geometry.type === 'LineString' || feature?.geometry.type === 'MultiLineString') {
+                    updatedLayers[LAYER_CYCLONE_TRACKS] = true;
+                }
+                return undefined;
+            });
+
+            stormPositions?.features?.find((feature) => {
+                if (feature?.geometry.type === 'Polygon' || feature?.geometry.type === 'MultiPolygon') {
+                    updatedLayers[LAYER_CYCLONE_BUFFERS] = true;
+                }
+                return undefined;
+            });
+
+            stormPositions?.features?.find((feature) => {
+                if (feature?.properties?.alert_level === 'Cones') {
+                    updatedLayers[LAYER_CYCLONE_UNCERTAINTY] = true;
+                }
+                return undefined;
+            });
```

Comment:
We don't need to mutate the updatedLayers. We should instead directly set these values to the object.